class-gf-razorpay.php

```diff
@@ -0,0 +1,285 @@
+<?php
+
+require_once ('razorpay-sdk/Razorpay.php');
+use Razorpay\Api\Api;
+
+GFForms::include_payment_addon_framework();
+
+class GFRazorpay extends GFPaymentAddOn {
+
+    protected $_version = GF_RAZORPAY_VERSION;
+    protected $_min_gravityforms_version = '1.9.3';
+    protected $_slug = 'razorpay-gravity-forms';
+    protected $_path = 'razorpay-gravity-forms/razorpay.php';
+    protected $_full_path = __FILE__;
+    protected $_url = 'http://www.gravityforms.com';
+    protected $_title = 'Gravity Forms Razorpay Add-On';
+    protected $_short_title = 'Razorpay';
+    protected $_supports_callbacks = true;
+
+    // Permissions
+    protected $_capabilities_settings_page = 'gravityforms_razorpay';
+    protected $_capabilities_form_settings = 'gravityforms_razorpay';
+    protected $_capabilities_uninstall = 'gravityforms_razorpay_uninstall';
+
+    // Automatic upgrade enabled
+    protected $_enable_rg_autoupgrade = true;
+
+    private static $_instance = null;
+
+    const GF_RAZORPAY_KEY = 'gf_razorpay_key';
+    const GF_RAZORPAY_SECRET = 'gf_razorpay_secret';
+
+    public static function get_instance()
+    {
+        if (self::$_instance == null)
+        {
+            self::$_instance = new GFRazorpay();
+        }
+
+        return self::$_instance;
+    }
+
+    public function init_frontend()
+    {
+        parent::init_frontend();
+
+        add_action('gform_after_submission', array($this, 'pay_using_razorpay'), 10, 2);
+    }
+
+    public function plugin_settings_fields()
+    {
+        return array(
+            array(
+                'title'       => '',
+                'fields'      => array(
+                    array(
+                        'name'    => self::GF_RAZORPAY_KEY,
+                        'label'   => esc_html__( 'Razorpay Key', 'razorpay-gravity-forms' ),
+                        'type'    => 'text',
+                        'class'   => 'medium',
+                    ),
+                    array(
+                        'name'    => self::GF_RAZORPAY_SECRET,
+                        'label'   => esc_html__( 'Razorpay Secret', 'razorpay-gravity-forms' ),
+                        'type'    => 'text',
+                        'class'   => 'medium',
+                    ),
+                    array(
+                        'type' => 'save',
+                        'messages' => array(
+                            'success' => esc_html__( 'Settings have been updated.', 'razorpay-gravity-forms' )
+                        ),
+                    ),
+                ),
```

Comment:
If this were code and multiple if blocks, I'd have agreed. This doesn't add complexity and is fairly readable.